tests/integration/components/paper/date-picker-test.js

```diff
@@ -0,0 +1,223 @@
+import hbs                        from 'htmlbars-inline-precompile';
+import { module, test }           from 'qunit';
+import { render, find, focus,
+  fillIn, blur, triggerKeyEvent } from '@ember/test-helpers';
+import { setupRenderingTest }     from 'ember-qunit';
+
+module('Integration | Component | {{paper/date-picker}}', function(hooks) {
+  setupRenderingTest(hooks);
+
+  hooks.beforeEach(function() {
+    this.set('value', new Date(2018, 9, 10, 0, 0));
+    this.actions = { update: (value) => {
+      this.set('value', value);
+    } };
+  });
+
+  test('date gets formatted to date string', async function(assert) {
+    this.set('value', new Date(2015, 0, 1));
+    await render(hbs`{{paper/date-picker value=value updated=(action "update")}}`);
+    let $input = find('input');
+    assert.equal($input.value, '01-01-15');
+  });
+
+  test('typing in an empty value', async function(assert) {
+    await render(hbs`{{paper/date-picker value=value updated=(action "update")}}`);
+
+    let $input = find('input');
+
+    await focus($input);
+    await fillIn($input, '');
+    await blur($input);
+
+    assert.equal(find('input').value, '');
+    assert.equal(this.get('value'), null);
+  });
+
+  test('filling in a date works', async function(assert) {
+    await render(hbs`{{paper/date-picker value=value updated=(action "update")}}`);
+
+    let $input = find('input');
+
+    await focus($input);
+    await fillIn(find('input'), '05-05-15');
+    await blur('input');
+
+    assert.equal(find('input').value, '05-05-15');
+    assert.equal(+this.get('value'), +(new Date(2015, 4, 5)));
+  });
+
+  test('typing in date shorthands', async function(assert) {
+    await render(hbs`{{paper/date-picker value=value updated=(action "update")}}`);
+
+    let $input  = find('input');
+    let year    = (new Date()).getFullYear();
+    let yearStr = year.toString().slice(-2);
+    let month   = `0${(new Date()).getMonth() + 1}`.substr(-2);
+
+    await focus($input);
+    await fillIn($input, '050515');
+    await blur($input);
+
+    assert.equal($input.value, '05-05-15');
+    assert.equal(+this.get('value'), +(new Date(2015, 4, 5)));
+
+    await focus($input);
+    await fillIn($input, '150515');
+    await blur($input);
+
+    assert.equal($input.value, '15-05-15');
+    assert.equal(+this.get('value'), +(new Date(2015, 4, 15)));
+
+    await focus($input);
+    await fillIn($input, '50515');
+    await blur($input);
+
+    assert.equal($input.value, '05-05-15');
+    assert.equal(+this.get('value'), +(new Date(2015, 4, 5)));
+
+    await focus($input);
+    await fillIn($input, '1505');
+    await blur($input);
+
+    assert.equal($input.value, `15-05-${yearStr}`);
```

Comment:
Put assert messages to all asserts in this test, so we know what is checked when we run / fail tests.